### PR TITLE
GIF images should not be scaled #8004

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/ImageServiceImpl.java [hunk 1]
```diff
@@ -53,67 +51,47 @@
     public ByteSource readImage( final ReadImageParams readImageParams )
         throws IOException
     {
-        final Path cachedImagePath = getCachedImagePath( readImageParams );
-        ByteSource imageByteSource = ImmutableFilesHelper.computeIfAbsent( cachedImagePath, () -> createImage( readImageParams ) );
-        return imageByteSource;
+        final String format = doGetFormatByMimeType( readImageParams.getMimeType() );
+
+        if ( renderAsASource( format ) )
```
Comment (anatol-sialitski) [inline modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/ImageServiceImpl.java:56]: I am not sure that `A` needs here, I think `renderAsSource` will be better

File: modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/ImageServiceImpl.java [hunk 2]
```diff
@@ -53,67 +51,47 @@
     public ByteSource readImage( final ReadImageParams readImageParams )
         throws IOException
     {
-        final Path cachedImagePath = getCachedImagePath( readImageParams );
-        ByteSource imageByteSource = ImmutableFilesHelper.computeIfAbsent( cachedImagePath, () -> createImage( readImageParams ) );
-        return imageByteSource;
+        final String format = doGetFormatByMimeType( readImageParams.getMimeType() );
```
Comment (rymsha) [inline modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/ImageServiceImpl.java:54]: It seems unnecessary to extract format from MIME type here as we only ignore processing very well defined `image/gif` MIME. So we can remove `renderAsASource` altogether and replace it with plain `"image/gif".equals(readImageParams.getMimeType())`

File: modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/ImageServiceImpl.java [hunk 3]
```diff
@@ -157,9 +135,6 @@ else if ( readImageParams.isScaleWidth() )
         //Filter string value
         final String filter = readImageParams.getFilterParam() != null ? readImageParams.getFilterParam() : "no-filter";
 
-        //Format string value
-        final String format = readImageParams.getFormat();
```
Comment (rymsha) [inline modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/ImageServiceImpl.java:161]: Move calculation of "format" here, so it is also guarded by cache.

File: modules/core/core-image/src/test/java/com/enonic/xp/core/impl/image/ImageServiceImplTest.java [hunk 4]
```diff
@@ -62,7 +60,7 @@ public void setUp()
         imageService.setImageFilterBuilder( imageFilterBuilder );
     }
 
-    @Test
+    /*@Test
```
Comment (rymsha) [inline modules/core/core-image/src/test/java/com/enonic/xp/core/impl/image/ImageServiceImplTest.java:63]: dead code. remove

**Pull request merged**